rules/prefer-dataset.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+const getDocsUrl = require('./utils/get-docs-url');
+
+const getMethodName = memberExpression => memberExpression.property.name;
+
+const getDataAttributeName = arg => {
+	if (arg.type === 'Literal') {
+		return (arg.value.match(/^data-(\w+)/) || ['', ''])[1];
+	}
+
+	return '';
+};
+
+const parseValueArgument = (context, arg) => {
+	return context.getSourceCode().getText(arg);
+};
+
+const getReplacement = (context, node, memberExpression, propertyName) => {
+	const objectName = memberExpression.object.name;
+	const value = parseValueArgument(context, node.arguments[1]);
+
+	return `${objectName}.dataset.${propertyName} = ${value}`;
+};
+
+const create = context => {
+	return {
+		CallExpression(node) {
+			const {callee} = node;
+
+			if (callee.type === 'MemberExpression' && getMethodName(callee) === 'setAttribute') {
```

Comment:
Code like `element[setAttribute]('data-unicorn', '🦄')` will "break" this check

Reply:
oh, I got you.
I have no idea which one is better...
- just ignore bracket notation
- fix it to element[dataset].unicorn = '🦄';

Reply:
You have to ignore bracket notation. It’s very hard to impossible to reliably check this.

Reply:
You should add some tests for this case too.